llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -905,6 +905,16 @@ Instruction *InstCombinerImpl::visitFMul(BinaryOperator &I) {
     }
   }
 
+  // (X * 0.0) * constant => X * 0.0
+  if (match(Op0, m_FMul(m_Value(X), m_AnyZeroFP())) &&
```

Comment:
Can you move this up, and combine with the existing handling of x * 0 -> copysign(0, x)